File: 2026-1/sesion-de-clase-03/dotnet/seguro-vehicular/Coberturas/CoberturaContraDesastresNaturales.cs
using Seguros;

namespace Coberturas;

public class CoberturaContraDesastresNaturales : SeguroDecorador
{
    public CoberturaContraDesastresNaturales(ICotizable seguro) : base(seguro)
    {
    }

    public override double CalcularCosto()
    {
        return this.seguro.CalcularCosto() + 300.00;
    }

    public override string Descripcion()
    {
        return this.seguro.Descripcion() + ", cobertura contra desastres naturales";
    }
}

File: 2026-1/sesion-de-clase-04/java/seguro-vehicular/pe/edu/softseguros/coberturas/CoberturaContraDesastresNaturales.java
package pe.edu.softseguros.coberturas;

import pe.edu.softseguros.modelo.Cotizable;

public class CoberturaContraDesastresNaturales extends SeguroDecorador {

    public CoberturaContraDesastresNaturales(Cotizable seguro) {
        super(seguro);
    }

    @Override
    public double CalcularCosto() {
        return this.seguro.CalcularCosto() + 300.00;
    }

    @Override
    public String Descripcion() {
        return this.seguro.Descripcion() + ", cobertura contra desastres naturales";
    }
}
